### ISSUE TITLE
Incompatibility with experimental Chrome "Tab Groups Save and Sync", restored suspended tabs are replaced with 'New Tab's

### ISSUE THREAD

**Issue by TheOverpassArsonist** [gioxx/MarvellousSuspender#229]
Please complete the following information when submitting a feature request or bug report.
 * Extension version: v7.1.6.2
 * Browser name & version: Brave v 1.62.156
 * Operating system & version: Linux, latest from arch repos

When using the chrome feature "Tab Groups Save and Sync" enablable in chrome::flags any suspended tabs that are part of a 'hidden' (which as far as I can tell means closed) group and then restored are replaced with new tabs. Non-suspended tabs are restored as-expected, but any suspended tabs are either not saved properly or not restored properly.

While suspended tabs do take less resources, being able to outright close tabs and restore them does still confer many advantages. Notably it makes organization much easier but even suspended tabs do still eat some resources so for managing lots of tabs being able to leverage both suspension and group saving would be desirable.

**Comment by mattdrewry** [comment]
Just to say I'm having problems with this too. Once I've clicked 'Save group' in Chrome I am unable to reactivate any tabs already sleeping, nor am I am to 'Suspend this tab now' as on click, nothing happens.

It would be great to be able to use both these feature side by side

* Extension version: v7.1.6.2
* Browser : Chrome Version 122.0.6261.95 (Official Build) (64-bit)
* Windows 11 Pro
  Version	23H2
  Installed on	‎06/‎10/‎2022
  OS build	22631.3155
  Experience	Windows Feature Experience Pack 1000.22684.1000.0

**Comment by msva** [comment]
By the way, chrome's group preview menu marks marvellous-suspended tabs as "Unsavable":
![Screenshot_20240309_203124](https://github.com/gioxx/MarvellousSuspender/assets/268102/bd0c7115-e7ca-4d65-af18-e197b2a1ad03)

**Comment by ZampolitGL** [comment]
Same here as [mattdrewry](https://github.com/mattdrewry) 
Extension version: v7.1.6.2
Browser : Chrome Version 123.0.6312.106 (Official Build) (64-bit)
Windows 10

**Comment by TheOverpassArsonist** [comment]
for what it's worth, in the issue which mentioned this one I discovered some oddities around this and found something that I think might be relevant
https://support.google.com/chrome/thread/256207412/tabs-of-local-pdf-files-are-unsavable-in-saved-groups?hl=en
in this issue on google someone notes that they had tabs of local files (i.e. : not websites) grouped that they typically closed and reopened, but once the tab saving feature was added now the tabs are marked as "unsavable" and open to new tabs as well. Since this is just a post on a google support form and not an actual issue request it's important to be mindful that what's said might not mean the exact words of what is being said, so I think what this is essentially saying is that they used to have many tabs of local files grouped, they would close their browser, (or maybe just click to fold/unfold the tabs?) then the next time they opened their browser they would reappear. Now however they noticed the tab save feature and using it turns their locally saved pdf files into new-tabs if they try to use that feature. (I believe they're likely conflating what they used to do with something else though, as closing the browser restores them fine and folding the group restores them fine) 

Testing myself I can confirm that local files are not saved in the tab groups "save" feature, indicating tab groups are unable to access local data for some reason, even if that data is still completely present

This problem is also getting significantly more severe because, while I originally enabled the feature in chrome::flags, now it appears to have officially rolled out and is enabled by default, meaning significantly more people will be unknowingly in danger of getting hit by this. Basically anyone who uses TMS and tab groups are likely to be hit eventually and since chrome (and all derivatives I am aware of) deceptively calls this feature "hiding", most will assume they're not doing anything destructive at all.

**Comment by Dibface** [comment]
Can confirm, this issue is still present. Closing and relaunching Chrome will delete all saved tabs within a group, replacing them with "New Tabs" instead (marked as "Unsavable Tab").

I'm seeing that there's 4 separate posts on this issue as well... and it seems to be a longstanding issue for a while now (as early as February???) 

Will this get fixed...?

**Comment by Technetium1** [comment]
No, it will almost certainly not be fixed. See more details: https://github.com/gioxx/MarvellousSuspender/issues/197#issuecomment-2185902490

**Comment by gioxx** [comment]
Flagging a possible connection: this and #238 (Google search page replacing suspended tabs in groups) may share the same root cause, Chrome's "Tab Groups Save and Sync" behavior interfering with TMS's suspended-tab URLs on restore, which a collaborator noted is likely not fixable on TMS's side (Chrome-side limitation). Not merging them since the exact symptom differs, but worth reading together.

**Comment by gioxx** [comment]
Correcting my previous comment here: re-reading this thread, the symptom described (grouped tabs marked "Unsavable" and replaced with "New Tab" on browser relaunch) looks like the same crbug.com/522338670 family that got fixed natively in Chrome 150 (shipped ~July 2026), not a separate, unfixable "Tab Groups Save and Sync" limitation as I said before, that assessment predates the fix. TMS's own workaround for the restart-replacement bug was dropped since Chrome now handles it correctly, and there's an on-demand "Repair grouped tab(s)" button on the Tab Health page as a fallback.

Could anyone still hitting this confirm whether it reproduces on a current Chrome (150+) with the latest TMS (v9.0.1)? Also relevant to #238.

**Comment by gioxx** [comment]
Closing as resolved. The root cause here: grouped suspended tabs coming back as "New Tab" / "Unsavable" after a browser relaunch, is crbug.com/522338670, which Chrome fixed natively in Chrome 150 (shipped ~July 2026). TMS's own workaround was removed since it's no longer needed, and the Tab Health page has an on-demand "Repair grouped tab(s)" button as a fallback.

No confirmation of a repro on Chrome 150+ with TMS v9.0.x in the ~month since the last check, so I'm closing this. If you're on a current Chrome and still see grouped suspended tabs break on restart, please comment and I'll reopen, that would point to something separate from the bug that got fixed.
